### DISCUSSION TITLE
What's the best way to load multiple images via folders into Nuxt/Image (Nuxt3)?

### DISCUSSION THREAD

**Discussion by Maximus1000** [nuxt/nuxt#16079] [Questions]
Hi All, 
Wondering if anyone can point me forward; thanks in advance for any advice.  Using Nuxt/Image (Nuxt3), I have folders of multiple images in the /public folder.  I'm trying to use the Node 'fs' package to read file names (server-side) and then provide that an array of filenames (client-side).  With that array, I pass to a lightbox component that will generate a gallery of images using.

I don't think there's an active dev gallery or lightbox viewer for Vue3 or Nuxt3 yet (noting several in Vue3 but nothing super active).  In client, I can provide an array of filenames statically to <nuxt-image> components, yet I'm trying to avoid that by using FS to read the file names.  My use-case is galleries of art images with items removed and added to subfolders often.  

I'm aware that reading file system on client-side presents security issues, so assuming that I've got to do this on server.  The approach below does in fact get filenames on server.  This array populates briefly in template but then is removed on final page load with error: Hydration text content mismatch.  I can't figure out the right way to grab hold of the filenames and persist that in client.  Lastly, I'm timid on how to perform tasks in /server as opposed to using process.server in template, noting that I may need to learn how that works.

This works in getting a list of file names but is removed during final client-side page load/hydration:  (pages/gallery.vue)
```
const fileNames = ref<any[]>([])
if(process.server) {
  const galleryPath = 'public/images/artist1/' //test
  const readDir = async (path:string) => {
    const files = fs.readdirSync(path, {withFileTypes: true})
      .filter(item => !item.isDirectory())
      .map(item => item.name)
    fileNames.value = files
  }
  readDir(galleryPath)
}
// a console.log of fileNames will show values but on final load, error.
// then, build up an object including full image paths to pass to gallery or lightbox component.
```
I tried using useState (haven't tried Pinia) as well as a watcher without success to persist from server to client.  This is my first time working with dynamic images, don't yet fully understand image providers as via nuxt/image.  Any advice appreciated!

edit: trying to build up gallery objects using fs (server-side) in a composable and a plugin, hadn't tried that approach.

Best
Marcus

**Comment by ryuhangyeong** [comment]
I also have similar requirements.

Maybe the link below will help you?

https://nuxt.com/docs/guide/directory-structure/server#server-directory

Here you can traverse the file with fs and return it in an array. And you can get it on the client side!
